File: vaccathon-rest-api/src/main/java/restapi/dto/Doctor.java
package restapi.dto;

import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Date;

@Document
public class Doctor extends Person{
    private String hospital;

    public Doctor(Long nationalInsuranceNumber, String firstName, String lastName, Date birthDay, String email, String phone, String city, String hospital) {
        super(nationalInsuranceNumber, firstName, lastName, birthDay, email, phone, city);
        this.hospital = hospital;
    }

    public String getHospital() {
        return hospital;
    }

    public Doctor setHospital(String hospital) {
        this.hospital = hospital;
        return this;
    }
}

File: frontend/vaccathon/src/app/models/doctor.ts
import {Person} from "./person";

export class Doctor extends Person {

  constructor(
    public  hospital: string,
    public  nationalInsuranceNumber: number,
    public  firstName: string,
    public  lastName: string,
    public  birthDay: Date,
    public  email: string,
    public  phone: string,
    public  city: string
  ) {
    super(nationalInsuranceNumber,
      firstName,
      lastName,
      birthDay,
      email,
      phone,
      city);
  }
}
